tests/test_ref.py

```diff
@@ -257,46 +257,17 @@ def test_nested_rgroup(self, spec):
         lhs = tuple(codec._unmap(msg))
         assert lhs == ((35, 'AE'),
                        (555, 1),
+                       (687, 'AA'),
                        (683, 2),
                        (688, '1'),
                        (689, '1'),
                        (688, '2'),
                        (689, '2'),
-                       (687, 'AA'),
-                       (17807, '11'),
-                       (10, '000')
-                       )
-        serialised = '35=AE;555=1;683=2;688=1;689=1;' \
-                     '688=2;689=2;687=AA;17807=11;10=000;'.replace(';', chr(1)).encode('UTF-8')
-        assert serialised == codec.serialise(msg)
-
-    def test_empty_rgroups(self, spec):
-        codec = Codec(spec=spec, decode_as='UTF-8')
-        msg = b'35=AJ;17807=11;232=2;233=bli;234=blu;' \
-              b'233=blih;234=bluh;555=0;10=000;'
-        msg = codec.parse(msg, separator=';')
-        assert {35: 'AJ',
-                17807: '11',
-                232: [
-                    {233: 'bli', 234: 'blu'},
-                    {233: 'blih', 234: 'bluh'}
-                ],
-                555: [],
-                10: '000'
-                } == msg
-        lhs = tuple(codec._unmap(msg))
-        assert lhs == ((35, 'AJ'),
-                       (232, 2),
-                       (233, 'bli'),
-                       (234, 'blu'),
-                       (233, 'blih'),
-                       (234, 'bluh'),
-                       (555, 0),
                        (17807, '11'),
                        (10, '000')
                        )
-        serialised = '35=AJ;232=2;233=bli;234=blu;233=blih;234=bluh;' \
-                     '555=0;17807=11;10=000;'.replace(';', chr(1)).encode('UTF-8')
+        serialised = '35=AE;555=1;687=AA;683=2;688=1;689=1;' \
+                     '688=2;689=2;17807=11;10=000;'.replace(';', chr(1)).encode('UTF-8')
         assert serialised == codec.serialise(msg)
 
```

Comment:
Why remove test function "test_empty_rgroups(self, spec)" ?

Reply:
no reason that I can remember, restored!

Reply:
A test function **test_from_dict_with_none()** is not found, also 4 lines of function **test_update_with_none()** is not found, I guess you made the change on an older version, maybe you should rebase the code based on the latest version.